### ISSUE TITLE
Niftool addon fails to install due to missing file Pyffi

### ISSUE THREAD

**Issue by Elliezrah** [niftools/blender_niftools_addon#411]
@niftools/blender-niftools-addon-reviewer -

Before creating a new issue, ensure that
 - [x] Read the user documentation. (Webpage to full documentation currently missing.)
 - [x] Check that the issue hasn't already been reported.
 - [ ] Read through the list of `Helpful resources` links on the right-hand side.
 - [ ] Talk with the developers to ensure its a valid issue

The issue will get closed off immediately as invalid if
 - [ ] Looking for support where the forums, chat or documentation is more appropriate.
 - [ ] Goes off topic, loses focus or turns into a discussion.
 - [ ] Removing parts of the template the issue.

Fill out the template below to the best of your ability, including .blend files, nifs, logs; help us to help you.

# Issue Overview
**[Provide a general introduction to the issue and why it is relevant to this repository]**

Hello, I've been trying very hard to get niftool working on blender.
I am getting a error log where it says Pyffi is not found. Stumbled across few forums where people with same issue.

https://www.blender.org/forum/viewtopic.php?p=73818

https://steamcommunity.com/app/365670/discussions/0/215439774861594828/

### Blender Niftools Addon Version Info
**[Provide version information for the Blender Niftools Addon the issue occurs with, available in Blender in the Add-On section]**

Blender 2.90, Niftool Addon v 0.003, 

### Blender Version Info
**[Provide information on which version of Blender the issue is seen]**

Blender 2.90

### Platform information
**[Provide a brief overview of what OS your running on]**

Windows 10 64bit

## Context
**[Provide more detailed information to the issue itself and why it is relevant]**

Can not proceed to use niftool.

## Steps to Reproduce
**[Ordered list of the steps required for recreating the issue, including settings]**
 #. 

Possibly happening for people who had not previously installed Python or Pyffi.
I have no knowledge about these softwares and I tried installing these two, getting it to work blindly. Still getting some blender upon addon installation.

**This is the error log before placing Pyffi files within blender**

```
Traceback (most recent call last):
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\modules\addon_utils.py", line 351, in enable
    mod = __import__(module_name)
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\addons\io_scene_niftools\__init__.py", line 64, in <module>
    import pyffi
ModuleNotFoundError: No module named 'pyffi'
```

**And this is the error log after placing Pyffi files within blender**

```
Traceback (most recent call last):
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\modules\addon_utils.py", line 351, in enable
    mod = __import__(module_name)
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\addons\io_scene_niftools\__init__.py", line 117, in <module>
    operators.register()
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\addons\io_scene_niftools\operators\__init__.py", line 42, in register
    from . import object, geometry, nif_import_op, nif_export_op, kf_import_op, egm_import_op  # kf_export_op
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\addons\io_scene_niftools\operators\nif_import_op.py", line 44, in <module>
    from io_scene_niftools.nif_import import NifImport
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\addons\io_scene_niftools\nif_import.py", line 42, in <module>
    import pyffi.spells.nif.fix
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\addons\pyffi\spells\nif\__init__.py", line 52, in <module>
    from pyffi.formats.nif import NifFormat
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\addons\pyffi\formats\nif\__init__.py", line 369, in <module>
    class NifFormat(FileFormat):
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\addons\pyffi\object_models\xml\__init__.py", line 98, in __init__
    xml_file = cls.openfile(xml_file_name, cls.xml_file_path)
  File "C:\Program Files\Blender Foundation\Blender 2.90\2.90\scripts\addons\pyffi\object_models\__init__.py", line 128, in openfile
    % (filename, filepaths))
OSError: 'nif.xml' not found in any of the directories [None, 'C:\\Program Files\\Blender Foundation\\Blender 2.90\\2.90\\scripts\\addons\\pyffi\\formats\\nif\\nifxml']
```

## Expected Result
**[Describe what you expected to have resulted from this process]**

Addon installs correctly.

## Actual Result
**[Describe what you currently experience from this process, and thereby explain the bug]**

Niftool plugin displays in addon list in blender but unable to activate. 

## Possible Fix
**[Optional, suggest fixes, improvement or reasons for the bug might have occurred]**
Possibly happening for people who had not previously installed Python or Pyffi.

## Screenshot
**[If relevant, include a screenshot]**

## Logs and Files
**[Provide logs file generated during the error as well as the blend and nif files are related to the issue]**

### Info Bar Output
**[Output from the Info View, available at top of Blender viewport, drag to expand]**

### Console Output
**[Set the logging level to 'Debug' and attach the output of the console. Enable via Window -> Toggle Console]**

### Blend File
**[Attach the blend file if the issue is reproducible]**

### Nif File
**[Attach input or output files, samples of what the expected output should be and reproducing the issue]**

## Similar Known Issues
**[Reference any known issues - https://github.com/niftools/blender_niftools_addon/issues]**

## Additional Information
**[Anything else you feel is relevant]**

**Comment by Elliezrah** [comment]
Much thanks for your assistance.

**Comment by neomonkeus** [comment]
I have seen the linked posts, they were just as wrong back then as they are not.

Step one - Delete all of the previous everythings: 
- If you have a folder called `io_scene_nif`, delete it
- If you have a folder called `io_scene_niftools`, delete it
- If you have any folders named `pyffi`, delete it.

Step two: 
- Use `blender_niftools_addon-v0.0.3-2020-12-13-704c40d.zipz` file as linked in the most recent release description - https://github.com/niftools/blender_niftools_addon/releases/tag/v0.0.3

Just to add to the above.
- The `Blender Niftools Add-on`, comes prepacked with all required dependencies.
- It contains the required version of Pyffi, there will never be a need to manually install it as an end user. (if you did want to, do it outside of Blender)
- I re-verified the install, it contains all of the requirements that you are missing as shown in this image: 
![image](https://user-images.githubusercontent.com/525919/106359686-1cddfe00-630c-11eb-8130-269cb1f3d356.png)

What I suspect happened is that didn't follow the release description and jumped to the download section and probably used the zip file named `Source`.
This is the source code, it is not the required built version, which doesn't include pyffi.
Likewise I suspect that you did the same when "installing" pyffi, which the source code is not the build one that is used byt the Blender Niftools Addon, so is missing dependencies which get included at build time, i.e 'nif.xml'

Hopefully cleaning out your addons directory and re-installing as per the guide provided in the release description will resolve your issue and you can use the auto-updater within the addon preference to update it in future and avoid having to worry about finding the latest versions.

**Comment by Elliezrah** [comment]
Hello Dear author,

I followed your instruction and it worked perfectly! Thank you so much.
Looks like my issue was that I unpacked the zip file and copied the files manually. I thought the file structure was weird for addon file so I copied "what I thought" to be only necessary files. 

Thank you again!

**Comment by neomonkeus** [comment]
Do you have any suggestion on how to improve the release description. 
We have added links, created gifs to try to avoid this problem, so if there is something obvious we are missing to make things easier, let us know.

**Comment by Elliezrah** [comment]
I viewed the file structure and I thought I had to unpack and copy the files separately. Because I am not a skilled user and I have installed addon for blender before but niftool file structure was different to others and unique. I did not know installing from within blender installation tool could just work it out fine. Maybe a bit of clarification for that could help other beginner level users like me.

Much thanks!
